snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/id/assigner/SnomedNamespaceAndModuleAssignerProvider.java

```diff
@@ -15,22 +15,35 @@
  */
 package com.b2international.snowowl.snomed.datastore.id.assigner;
 
-import javax.inject.Provider;
+import java.util.Map;
 
 import com.b2international.commons.platform.Extensions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMap.Builder;
 
 /**
  * @since 6.3
  */
-public enum SnomedNamespaceAndModuleAssignerProvider implements Provider<SnomedNamespaceAndModuleAssigner> {
+public enum SnomedNamespaceAndModuleAssignerProvider {
 
 	INSTANCE;
 
 	private static final String NAMESPACE_ASSIGNER_EXTENSION = "com.b2international.snowowl.snomed.datastore.snomedNamespaceAndModuleAssigner";
 	
-	@Override
-	public SnomedNamespaceAndModuleAssigner get() {
-		return Extensions.getFirstPriorityExtension(NAMESPACE_ASSIGNER_EXTENSION, SnomedNamespaceAndModuleAssigner.class);
+	private final Map<String, SnomedNamespaceAndModuleAssigner> assigners;
+	
+	private SnomedNamespaceAndModuleAssignerProvider() {
+		Builder<String, SnomedNamespaceAndModuleAssigner> mapBuilder = ImmutableMap.builder();
```

Comment:
`ImmutableMap.builder` throws an exception when tries to register another assigner with an already registered name. It will fail to initialize the class, which leads to `NoClassDefFoundError` and Snow Owl startup error. This registration phase should report duplicate assigners as warnings in the log and should only register the first assigner.